pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
+	// Result of validation performed at ValidatedOn.
+	ValidationResult SignatureValidationResult
+
+	// At most one of the following pointers can point to details. If the signature type isn't recognized,
+	// they will all be nil.
+
+	// Details for signature of type "atomic".
+	AtomicSignatureDetails *AtomicSignatureDetails
+	// For any future supported signature type, add additional pointer to its own details struct.
+}
+
+// AtomicSignatureDetails holds metadata of recognized signature of type "atomic". They are parsed from
+// signature's content.
+type AtomicSignatureDetails struct {
+	// Digest of signed manifest.
+	SignedImageName string
+	// Docker image reference
+	SignedDockerImageReference string
```

Comment:
> Don't use ImageReference, since in the API that means something different.

(I have no opinion on OpenShift API naming.

Also, the design as is uses a single “docker-reference” namespace for both plain Docker registry repositories and OpenShift imagestreams. It seems safe enough to expect that there will not be an OpenShift registry and an independent Docker registry on the same host name, and having only one field name / namespace simplifies things, but I wanted to highlight this in case there were any problems with this that I haven’t noticed.)